jcli/src/jcli_app/certificate/new_vote_cast.rs

```diff
@@ -1,44 +1,106 @@
 use crate::jcli_app::certificate::{write_cert, Error};
+use crate::jcli_app::utils;
+use bech32::FromBase32;
 use chain_impl_mockchain::{
     certificate::{Certificate, VoteCast, VotePlanId},
     vote::{Choice, Payload},
 };
+use rand_chacha::rand_core::SeedableRng;
 use std::path::PathBuf;
 use structopt::StructOpt;
 
-/// create a vote cast certificate
 #[derive(StructOpt)]
-pub struct VoteCastCmd {
+pub struct PublicVoteCast {
     /// the vote plan identified on the blockchain
-    pub vote_plan_id: VotePlanId,
+    vote_plan_id: VotePlanId,
 
     /// the number of proposal in the vote plan you vote for
-    pub proposal_index: u8,
+    proposal_index: u8,
 
     /// the number of choice within the proposal you vote for
-    pub choice: u8,
+    choice: u8,
```

Comment:
Let's give these `structopt` annotation to make them options instead of positional parameters?
Or maybe just the first: `--vote-plan-id=foo 5 1` is mnemonic enough.